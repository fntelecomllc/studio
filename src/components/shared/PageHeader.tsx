
import type { LucideIcon } from 'lucide-react';
import React from 'react';

interface PageHeaderProps {
  title: string;
  description?: string;
  icon?: LucideIcon;
  actionButtons?: React.ReactNode;
}

export default function PageHeader({ title, description, icon: Icon, actionButtons }: PageHeaderProps) {
  return (
    <div className="mb-6 flex flex-col sm:flex-row items-start sm:items-center justify-between gap-4">
      <div className="flex items-center gap-3">
        {Icon && <Icon className="h-7 w-7 text-primary" />}
        <div>
          <h1 className="text-2xl font-bold tracking-tight text-foreground sm:text-3xl">{title}</h1>
          {description && <p className="text-muted-foreground mt-1">{description}</p>}
        </div>
      </div>
      {actionButtons && <div className="flex gap-2 self-end sm:self-auto">{actionButtons}</div>}
    </div>
  );
}
