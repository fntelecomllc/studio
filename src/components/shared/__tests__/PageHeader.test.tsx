import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PageHeader from '../PageHeader';
import { Home } from 'lucide-react';

describe('PageHeader', () => {
  it('renders the title', () => {
    render(<PageHeader title="Test Title" />);
    expect(screen.getByText('Test Title')).toBeInTheDocument();
  });

  it('renders the description when provided', () => {
    render(<PageHeader title="Test Title" description="Test Description" />);
    expect(screen.getByText('Test Description')).toBeInTheDocument();
  });

  it('does not render description when not provided', () => {
    render(<PageHeader title="Test Title" />);
    expect(screen.queryByText('Test Description')).not.toBeInTheDocument();
  });

  it('renders the icon when provided', () => {
    render(<PageHeader title="Test Title" icon={Home} />);
    // Check if an SVG is rendered (Lucide icons are SVGs)
    const svg = document.querySelector('svg');
    expect(svg).toBeInTheDocument();
  });

  it('renders action buttons when provided', () => {
    const ActionButton = () => <button>Click Me</button>;
    render(<PageHeader title="Test Title" actionButtons={<ActionButton />} />);
    expect(screen.getByText('Click Me')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <PageHeader 
        title="Snapshot Title" 
        description="Snapshot Description" 
        icon={Home}
        actionButtons={<button>Action</button>}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
