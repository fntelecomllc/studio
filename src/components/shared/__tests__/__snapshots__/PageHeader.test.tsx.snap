// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader matches snapshot 1`] = `
<div>
  <div
    class="mb-6 flex flex-col sm:flex-row items-start sm:items-center justify-between gap-4"
  >
    <div
      class="flex items-center gap-3"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-house h-7 w-7 text-primary"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
        />
        <path
          d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
        />
      </svg>
      <div>
        <h1
          class="text-2xl font-bold tracking-tight text-foreground sm:text-3xl"
        >
          Snapshot Title
        </h1>
        <p
          class="text-muted-foreground mt-1"
        >
          Snapshot Description
        </p>
      </div>
    </div>
    <div
      class="flex gap-2 self-end sm:self-auto"
    >
      <button>
        Action
      </button>
    </div>
  </div>
</div>
`;
